vignettes/speedup_options.Rmd.orig

```diff
@@ -0,0 +1,369 @@
+```{r crps-func}
+# A function to calculate the CRPS
+calc_crps <- function(x, truth) {
+  shortest_obs_length <- min(ncol(x), length(truth))
+  reduced_truth <- tail(truth, shortest_obs_length)
+  reduced_x <- tail(t(x), shortest_obs_length)
+  return(crps_sample(reduced_truth, reduced_x))
+}
+```
+
+Now, we can calculate CRPS for the Rt and infection estimates of the various models and using `calc_crps()`.
+
+```{r fit_crps}
+# Get the Rt samples
+Rt_estimated <- lapply(results, function(x) {
+  if ("R[1]" %in% names(x$estimates$fit)) {
+    extract(x$estimates$fit, "R")$R
+  } else {
+    extract(x$estimates$fit, "gen_R")$gen_R
+  }
+})
+# CRPS for the Rt estimates
+rt_crps <- lapply(
+  Rt_estimated,
+  calc_crps,
+  truth = R
+)
+
+# Get the infection samples
+infections_estimated <- lapply(results, function(x) {
+  extract(x$estimates$fit, "infections")$infections
+})
+
+# CRPS for the infections estimates
+infections_crps <- lapply(
+  infections_estimated,
+  calc_crps,
+  truth = infections_true
+)
+```
+
+We will now post-process the CRPS results to make them easier to visualise.
+
+```{r fit_postprocessing}
+# Post-processing the CRPS results of the Rt estimates
+rt_df <- as.data.table(rt_crps)
+rt_df[, metric := "CRPS"]
+rt_df[, time := 1:.N]
+rt_df <- melt(
+  rt_df,
+  id.vars = c("metric", "time"),
+  variable.name = "model"
+)
+
+# Post-processing the CRPS results of the infection estimates
+infections_df <- as.data.table(infections_crps)
+infections_df[, metric := "CRPS"]
+infections_df[, time := 1:.N]
+infections_df <- melt(
+  infections_df,
+  id.vars = c("metric", "time"),
+  variable.name = "model"
+)
+```
+
+Let's visualise the CRPS results for the Rt and infection estimates.
+
+```{r rt_plot}
+rt_plot <- ggplot(rt_df, aes(x = time, y = value, colour = model)) +
+  geom_line() +
+  scale_colour_brewer("Model", palette = "Dark2") +
+  xlab("Time") +
+  ylab("CRPS") +
+  ggplot2::theme_bw() +
+  ggtitle("Reconstructing R")
+plot(rt_plot)
+```
+
+```{r infections_plot}
+infections_plot <- ggplot(infections_df, aes(x = time, y = value, colour = model)) +
+  geom_line() +
+  scale_colour_brewer("Model", palette = "Dark2") +
+  xlab("Time") +
+  ylab("CRPS") +
+  ggplot2::theme_bw() +
+  ggtitle("Reconstructing infections")
+plot(infections_plot)
+```
+
+## Some considerations
+
+### Mechanistic vs non-mechanistic models
+
+- Estimation in _EpiNow2_ using the mechanistic approaches (prior on Rt) is often much slower than the non- mechanistic approach. The mechanistic model is slower because it models intricate details about the processes and mechanisms that drive $Rt$ estimates. The nonmechanistic model, on the other hand, is faster because it does not model these intricate details. If performing a retrospective analysis, the _non-mechanistic model_ (where `rt = NULL`) is better suited and a faster alternative to the mechanistic $Rt$ model.
```

Comment:
```suggestion
- Estimation in _EpiNow2_ using the mechanistic approaches (prior on Rt) is often much slower than the non-mechanistic approach. The mechanistic model is slower because it models intricate details about the processes and mechanisms that drive $Rt$ estimates. On the other hand, the non-mechanistic model (where `rt = NULL`) is faster because it does not model these intricate details. If performing a retrospective analysis, the _non-mechanistic model_ is better suited and a faster alternative to modelling $Rt$.
```